src/instructlab/sdg/pipeline.py

```diff
@@ -184,9 +188,12 @@ def _generate_single(self, dataset) -> Dataset:
 
             # If at any point we end up with an empty data set, the pipeline has failed
             if len(dataset) == 0:
-                raise EmptyDatasetError(
```

Comment:
How about we just allow `generate()` to return empty without error (i.e. it would just return here) and move the error checking into `generate_data()`. That way we can do it differently for mmlubench?

Reply:
will do